File: restdoc-client/src/main/java/restdoc/client/remoting/PostEmptyApiTemplateRequestProcessor.java
package restdoc.client.remoting;

import io.netty.channel.ChannelHandlerContext;
import org.springframework.beans.factory.annotation.Autowired;
import restdoc.client.context.EndpointsListener;
import restdoc.remoting.common.body.GetClientApiListRequestBody;
import restdoc.remoting.netty.NettyRequestProcessor;
import restdoc.remoting.protocol.RemotingCommand;
import restdoc.remoting.protocol.RemotingSysResponseCode;

public class PostEmptyApiTemplateRequestProcessor implements NettyRequestProcessor {

    private final EndpointsListener endpointsListener;

    @Autowired
    public PostEmptyApiTemplateRequestProcessor(EndpointsListener endpointsListener) {
        this.endpointsListener = endpointsListener;
    }

    @Override
    public RemotingCommand processRequest(ChannelHandlerContext ctx, RemotingCommand request) throws Exception {
        RemotingCommand response = RemotingCommand.createResponseCommand(RemotingSysResponseCode.SUCCESS,
                "OK");

        GetClientApiListRequestBody body = new GetClientApiListRequestBody();
        body.setApiEmptyTemplates(endpointsListener.getEmptyApiTemplates());

        response.setBody(body.encode());

        return response;
    }

    @Override
    public boolean rejectRequest() {
        return false;
    }
}

File: restdoc-rpc-client/restdoc-restweb-client/src/main/java/restdoc/client/remoting/PostEmptyApiTemplateRequestProcessor.kt
package restdoc.client.remoting

import io.netty.channel.ChannelHandlerContext
import org.springframework.beans.factory.annotation.Autowired
import restdoc.client.context.EndpointsListener
import restdoc.remoting.common.body.RestWebExposedAPIBody
import restdoc.remoting.netty.NettyRequestProcessor
import restdoc.remoting.protocol.RemotingCommand
import restdoc.remoting.protocol.RemotingSysResponseCode

@Deprecated("")
class PostEmptyApiTemplateRequestProcessor @Autowired constructor(private val endpointsListener: EndpointsListener) : NettyRequestProcessor {
    @Throws(Exception::class) override fun processRequest(ctx: ChannelHandlerContext, request: RemotingCommand): RemotingCommand {
        val response = RemotingCommand.createResponseCommand(RemotingSysResponseCode.SUCCESS,
                "OK")
        val body = RestWebExposedAPIBody()
        body.apiList = endpointsListener.emptyApiTemplates
        response.body = body.encode()
        return response
    }

    override fun rejectRequest(): Boolean {
        return false
    }

}